sw/device/lib/crypto/include/ecc.h

```diff
@@ -107,12 +107,19 @@ typedef struct ecc_curve {
 /**
  * Performs the key generation for ECDSA operation.
  *
- * Computes private key (d) and public key (Q) keys for ECDSA
- * operation.
+ * Computes private key (d) and public key (Q) keys for ECDSA operation.
  *
- * The domain_parameter field of the `elliptic_curve` is required
- * only for a custom curve. For named curves this field is ignored
- * and can be set to NULL.
+ * The caller should allocate and partially populate the blinded key struct,
+ * including populating the key configuration and allocating space for the
+ * keyblob. The caller should indicate the length of the allocated keyblob;
+ * this function will return an error if the keyblob length does not match
+ * expectations. For hardware-backed keys, the keyblob length is 0 and the
+ * keyblob pointer may be NULL. For non-hardware-backed keys, the keyblob
+ * should be twice the length of the key. The value in the `checksum` field of
+ * the blinded key struct will be populated by the key generation function.
+ *
+ * The domain_parameter field of the `elliptic_curve` is required only for a
+ * custom curve. For named curves this field is ignored and can be set to NULL.
  *
  * @param elliptic_curve Pointer to the elliptic curve to be used
  * @param private_key Pointer to the blinded private key (d) struct
```

Comment:
```suggestion
 * @param[out] private_key Pointer to the blinded private key (d) struct
```
This applies for the rest of the arguments documentation in this file.